core/arch/arm/crypto/aes-gcm-ce.c

```diff
@@ -81,6 +81,67 @@ void internal_aes_gcm_ghash_update(struct internal_aes_gcm_ctx *ctx,
 }
 
 #ifdef ARM64
+static uint32_t ror32(uint32_t word, unsigned int shift)
+{
+	return (word >> shift) | (word << (32 - shift));
+}
+
+TEE_Result internal_aes_gcm_expand_enc_key(const void *key, size_t key_len,
```

Comment:
Why ARM64 only?

I understand you're copying code from `core/lib/libtomcrypt/src/ciphers/` (`aes_armv8a_ce.c` and `aes_modes_armv8a_ce_a64.S`) into `core/arch/arm/crypto/` (`aes-gcm-ce.c` and `ghash-ce-core_a64.S`) to cut that dependency on LTC so that pager can use accelerated code. Two things:

1. You should clearly say where the code comes from in the commit log
2. Why not update similarly `ghash-ce-core_a32.S` using code from `aes_modes_armv8a_ce_a32.S`(function `ce_aes_sub()`) ?

Reply:
1. Yes, I'll update the commit message
2. We don't have any AES routines for ARM32 yet so there's nothing to be done in this area for ARM32 yet.

Reply:
`core/lib/libtomcrypt/src/ciphers/aes_modes_armv8a_ce_a32.S`?

Reply:
Sorry I wasn't clear. In ARM64 we have an implementation with AES and GHASH fused, which gives a good speedup. In ARM32 we don't have such an implementation (yet, ever?) so instead we're using the AES API in the cryptolib.

What we could do, is to extract the CE AES implementation from LTC and supply it to LTC similarly to how GHASH is supplied. The advantage is that we can get rid of some redundancy and easier access to the routines. This however is not in that scope of this PR.

Reply:
I think I've got it now.